addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/manager/DeviceStructureManager.java

```diff
@@ -0,0 +1,290 @@
+     *
+     * @throws IOException
+     * @throws ApiException
+     */
+    private void refreshDevices() throws IOException, ApiException {
+        List<Device> devices = client.getFullDevices();
+        for (Device d : devices) {
+            handleRefreshedDevice(d);
+        }
+    }
+
+    /**
+     * Refreshs the {@link Device} with the given id and stores it in the {@link DeviceStructureManager}.
+     *
+     * @param deviceId
+     * @throws IOException
+     * @throws ApiException
+     */
+    public void refreshDevice(String deviceId) throws IOException, ApiException {
+        Device d = client.getFullDeviceById(deviceId);
+        handleRefreshedDevice(d);
+    }
+
+    /**
+     * Stores the newly refreshed {@link Device} in the {@link DeviceStructureManager} structure and logs the
+     * {@link Device}s details and state, if the debug logging is enabled.
+     *
+     * @param d the {@link Device}
+     */
+    private void handleRefreshedDevice(Device d) {
+        if (InnogyBindingConstants.SUPPORTED_DEVICES.contains(d.getType())) {
+            addDeviceToStructure(d);
+        } else {
+            logger.debug("Device {}:'{}' by {} ({}) ignored - UNSUPPORTED.", d.getType(), d.getName(),
+                    d.getManufacturer(), d.getId());
+            logger.debug("====================================");
+            return;
+        }
+
+        if (d.isController()) {
+            bridgeDeviceId = d.getId();
+        }
+
+        if (logger.isDebugEnabled()) {
+            try {
+                logger.debug("Device {}:'{}' by {} ({}) loaded.", d.getType(), d.getName(), d.getManufacturer(),
+                        d.getId());
+                for (Capability c : d.getCapabilityMap().values()) {
+                    logger.debug("> CAP: {}/{} ({})", c.getType(), c.getName(), c.getId());
+                    if (d.isRadioDevice() && !d.isReachable()) {
+                        logger.debug(">> CAP-State: unknown (device NOT REACHABLE).");
+                    } else {
+                        if (c.getCapabilityState() != null) {
+                            for (Property p : c.getCapabilityState().getStateMap().values()) {
+                                logger.debug(">> CAP-State: {} -> {} ({})", p.getName(), p.getValue(),
+                                        p.getLastchanged());
+                            }
+                        } else {
+                            logger.debug(">> CAP-State: unknown (NULL)");
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("EX: ", e);
+            }
+            logger.debug("====================================");
+        }
+    }
+
+    /**
+     * Writes an exemplary openHAB configuration for the found {@link Device}s to the logfile
+     * which then can be copied to the config files easily for initial use of the things.
+     */
+    private void writeExampleConfigToLog() {
```

Comment:
As mentioned above, please remove this "feature".
If you think this is a useful feature, let us discuss an PR against ESH that brings a console command with which inbox results & existing things can be dumped out in things-file-syntax.